src/api_generation/high_level.jl

```diff
@@ -109,7 +134,8 @@ function _generate_high_level_definition(
     """
     Generate function definition for a service request given required, header and idempotent parameters.
     """
-    function _generate_rest_operation_defintion(
+    function _generate_operation_definition(
+        protocol::RestBased,
```

Comment:
Seems like we don't really need an instance at all can just do 
```suggestion
        protocol::Type{RestBased},
```

Reply:
Instances are better for the compiler and easier to read.